src/core/MOM_dynamics_split_RK2.F90

```diff
@@ -860,6 +888,93 @@ subroutine step_MOM_dyn_split_RK2(u, v, h, tv, visc, Time_local, dt, forces, p_s
   if (CS%id_u_BT_accel > 0) call post_data(CS%id_u_BT_accel, CS%u_accel_bt, CS%diag)
   if (CS%id_v_BT_accel > 0) call post_data(CS%id_v_BT_accel, CS%v_accel_bt, CS%diag)
 
+  ! Diagnostics for terms multiplied by fractional thicknesses
+  do j=js,je ; do I=Isq,Ieq
+    hf_PFu_2d(I,j) = 0.0
+    hf_CAu_2d(I,j) = 0.0
+    hf_uBT_accel_2d(I,j) = 0.0
+  enddo ; enddo
+  do J=Jsq,Jeq ; do i=is,ie
+    hf_PFv_2d(i,J) = 0.0
+    hf_CAv_2d(i,J) = 0.0
+    hf_vBT_accel_2d(i,J) = 0.0
+  enddo ; enddo
+
```

Comment:
These should be guarded behind `if (CS%id_PFu_2d>0)` flags. Also explicit loops are not required when initializing arrays.

Reply:
Yes, I am going to add this.